packages/browser/src/plugins/env-enrichment/index.ts

```diff
@@ -198,29 +143,9 @@ class PageEnrichmentPlugin implements Plugin {
     const event = ctx.event
     const evtCtx = (event.context ??= {})
 
-    const defaultPageContext = pageDefaults()
-
-    let pageContextFromEventProps: Pick<EventProperties, string> | undefined
-
-    if (event.type === 'page') {
-      pageContextFromEventProps =
-        event.properties &&
-        pick(event.properties, Object.keys(defaultPageContext))
-
-      event.properties = {
-        ...defaultPageContext,
-        ...event.properties,
-        ...(event.name ? { name: event.name } : {}),
-      }
-    }
-
-    evtCtx.page = {
-      ...defaultPageContext,
-      ...pageContextFromEventProps,
-      ...evtCtx.page,
-    }
-
-    const query: string = evtCtx.page.search || ''
+    const search = evtCtx.page!.search || ''
```

Comment:
Note: `page` should never be empty since the page is created as part of the event factory. We probably can/should update the SegmentEvent type to reflect this.

Reply:
If we have an integration test that can guarantee that we don't move this order I think this is fine. If we don't then it might be better to add the extra lines to check it and prevent a crash.

Reply:
dit: Seth response (github bug, I'm not sure why I can't comment)
> If we have an integration test that can guarantee that we don't move this order I think this is fine. If we don't then it might be better to add the extra lines to check it and prevent a crash.

That's a good call out! IMO, there would be a runtime error in all of our integration tests here if page was not an object, so I actually felt OK about not adding a specific test. I couldn't think of a scenario where you could make that mistake in a refactor, unless you were to eliminate all integration tests. Typically I do agree that having an explicit test is better than implicit, and I could go either way!